lib/galaxy/datatypes/sniff.py

```diff
@@ -723,18 +734,15 @@ def handle_compressed_file(
         with tempfile.NamedTemporaryFile(prefix=tmp_prefix, dir=tmp_dir, delete=False) as uncompressed:
             compressed_file = DECOMPRESSION_FUNCTIONS[compressed_type](filename)
```

Comment:
I'd remove file handle closing from `file_reader()` and use a `with` statement:

```suggestion
            with DECOMPRESSION_FUNCTIONS[compressed_type](filename) as compressed_file:
```

?

Reply:
Yep, that looks better!